service/middleware.go

```diff
@@ -174,12 +174,19 @@ func (uis *UIServer) isSuperUser(u gimlet.User) bool {
 
 func (uis *UIServer) setCORSHeaders(next http.HandlerFunc) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		if r.Method == http.MethodGet && len(uis.Settings.Ui.CORSOrigins) > 0 {
+		if len(uis.Settings.Ui.CORSOrigins) > 0 {
 			requester := r.Header.Get("Origin")
+
+			// Requests from a GQL client include this header, which must be added to the response to enable CORS
+			gqlHeader := r.Header.Get("Access-Control-Request-Headers")
+
 			if util.StringSliceContains(uis.Settings.Ui.CORSOrigins, requester) {
 				w.Header().Add("Access-Control-Allow-Origin", requester)
 				w.Header().Add("Access-Control-Allow-Credentials", "true")
 				w.Header().Add("Access-Control-Allow-Headers", fmt.Sprintf("%s, %s", evergreen.APIKeyHeader, evergreen.APIUserHeader))
+				if gqlHeader != "" {
```

Comment:
This will overwrite the headers in the line above, these all should be added at once

Reply:
nice! changed it